force-app/main/interoperability/classes/ProductRegistryVMapper.cls

```
@@ -0,0 +1,87 @@
+     * @description Empty constructor is not public for Singleton Pattern
+     **********************************************************************/
+    protected ProductRegistryVMapper() {
+    }
+
+    /*********************************************
+     * @description Instance for Singleton Pattern
+     **********************************************/
+    public static ProductRegistryVMapper getInstance() {
+        if (instance == null) {
+            instance = new ProductRegistryVMapper();
+        }
+
+        return instance;
+    }
+
+    /********************************************************************************************
+     * @description Returns the list of ProductRegistryVModel based on the list of ProductRegistryModel
+     * @return   A list of ProductRegistryVModel with all its properties populated
+     ********************************************************************************************/
+    public virtual List<ProductRegistryVModel> getProductRegistrySettingsProductInformation() {
+        List<ProductRegistryVModel> productRegistryVModelList = new List<ProductRegistryVModel>();
+        List<ProductRegistryModel> productRegistryModelList = new List<ProductRegistryModel>();
+
+        productRegistryModelList = this.locateProductRegistryService().getProductRegistrySettingsProductInformation();
+
+        if (productRegistryModelList != null && !productRegistryModelList.isEmpty()) {
+            for (ProductRegistryModel prodRegistryModel : productRegistryModelList) {
+                ProductRegistryVModel prodRegistryVModel = new ProductRegistryVModel(
+                    prodRegistryModel.action,
+                    prodRegistryModel.namespace,
+                    prodRegistryModel.classname,
+                    prodRegistryModel.apiVersion
+                );
+                productRegistryVModelList.add(prodRegistryVModel);
+            }
+        }
```

This is another place where we can make it abundantly clear what we expect from a unit testing perspective (and where another use case was hiding). Consider whether or not productRegistryModelList can ever be null, and if it can't, then we don't need to account for that use case. Though if it is, do we want to throw an exception?

If the list is empty, then the for loop won't do much harm from a processing perspective, since it will exit before it begins. So we don't need to list it in this case.

```suggestion
        if (productRegistryModelList == null) {
            return null;
        }

        for (ProductRegistryModel prodRegistryModel : productRegistryModelList) {
            ProductRegistryVModel prodRegistryVModel = new ProductRegistryVModel(
                prodRegistryModel.action,
                prodRegistryModel.namespace,
                prodRegistryModel.classname,
                prodRegistryModel.apiVersion
            );
            productRegistryVModelList.add(prodRegistryVModel);
        }
```